src/tcp/tcp-socket.js

```diff
@@ -106,6 +113,33 @@ TcpSocket.prototype._onData = function( packet ) {
 };
 
 /**
+ * When the message recieved is greater than the maximum allowed amount specified
+ * va maxMessageSize this will remove the preceding valid messages in the buffer
+ * or dump the buffer and emit a MAXIMUM_MESSAGE_SIZE_EXCEEDED event.
+ * 
+ *
+ * @emits 	{String} message
+ *
+ * @private
+ * @returns {void}
+ */
+TcpSocket.prototype._maxMessageSizeExceed = function() {
+	var errorMsg, indexEOM;
+
```

Comment:
I feel that this should be on the connection layer as both engine.io and TCP are affected